packages/fuel-gauge/src/contract.test.ts

```diff
@@ -1266,4 +1268,38 @@ describe('Contract', () => {
     expect(scriptGasLimit?.toNumber()).toBe(gasLimit);
     expect(bn(maxFeePolicy?.data).toNumber()).toBe(maxFee);
   });
+
+  it('can call SMO contract', async () => {
+    using launched = await launchTestNode({
+      contractsConfigs: [
+        {
+          factory: SmoContractFactory,
+        },
+      ],
+    });
+
+    const {
+      provider,
+      wallets: [recipient],
+      contracts: [contract],
+    } = launched;
+
+    const data = true;
+    const baseAssetId = provider.getBaseAssetId();
+
+    const { waitForResult } = await contract.functions
+      .send_typed_message_bool(recipient.address.toB256(), data, 1)
+      .callParams({ forward: [1, baseAssetId] })
+      .call();
+
+    const {
+      transactionResult: { receipts },
+    } = await waitForResult();
+
+    const messageOutReceipt = receipts.find(
+      ({ type }) => ReceiptType.MessageOut === type
+    ) as ReceiptMessageOut;
+
+    expect(messageOutReceipt.recipient).toBe(recipient.address.toB256());
```

Comment:
I suppose that the `data` field of the [`MessageOut` receipt](https://docs.fuel.network/docs/specs/abi/receipts/#messageout-receipt) is tied to its respective entry in the `messageTypes` array of the abi. It seems to me that part of supporting them, as the linked issue #2347 requires, would be to have this data decoded in these receipts.
```ts
expect(messageOutReceipt.data).toStrictEqual(myData);
```

Reply:
@nedsalk The [approach](https://github.com/FuelLabs/fuels-ts/pull/3016/commits/aa7dbd24f3efa7b4ffe05bb7f901d52546642174) that I am using to decode the message data has worked for both `boolean` and `u8` Sway types. However, it does not work for the [Bytes](https://github.com/FuelLabs/fuels-ts/pull/3016/commits/4ebad43c67b0bcd60f1e1e3eb71ac566786d3d9d) type. What I am missing? This is the [contract](https://github.com/FuelLabs/fuels-ts/pull/3016/commits/7771e1a10a34e16863c10f27a5ac4734554e98c8) and its [JSON ABI](https://gist.github.com/Torres-ssf/99c16683949cd69b03fef9c4f443c76f)

Reply:
@Torres-ssf yeah, we can put it in another issue, the problem is more complex than we thought initially.

Reply:
@nedsalk Feel free to edit the description and add more context if you would like.